src/net_processing.cpp

```diff
@@ -947,10 +949,10 @@ bool PeerManagerImpl::TipMayBeStale()
 {
     AssertLockHeld(cs_main);
     const Consensus::Params& consensusParams = m_chainparams.GetConsensus();
-    if (m_last_tip_update == 0) {
-        m_last_tip_update = GetTime();
+    if (count_seconds(m_last_tip_update) == 0) {
+        m_last_tip_update = GetTime<std::chrono::seconds>();
     }
-    return m_last_tip_update < GetTime() - consensusParams.nPowTargetSpacing * 3 && mapBlocksInFlight.empty();
+    return count_seconds(m_last_tip_update) < GetTime() - consensusParams.nPowTargetSpacing * 3 && mapBlocksInFlight.empty();
```

Comment:
I think it would have been better to use type-safe time here (the goal of this pull), because:

* GetTime should eventually be removed
* count_seconds should only be used to "serialize" time values (for example in RPC results or P2P messages)

(see comments in `src/util/time.h`)

Reply:
> I think it would have been better to use the type-safe time here

My first approach was to do precisely this. But since **m_last_tip_update** was `atomic<seconds>` type.
There were two issues I faced:
1. Converting RHS to match the type
2. For some reason, subtraction between atomic types is not possible

I tried to find some way to convert `atomic<seconds>` to `<seconds>`, but I could not find a satisfactory result. That's why I went with the current implementation.

Reply:
I think `m_last_tip_update.load()` will load the value wrapped by the atomic

Reply:
> I think m_last_tip_update.load() will load the value wrapped by the atomic

It worked! Let me open PR with all the suggested changes, and wrap up this chain of PR of going from `int` -> `chrono` as soon as possible. 

Thanks a lot!